src/App.js

```diff
@@ -1,109 +1,166 @@
-                - {value.toLocaleString('de-DE')} watt/h - increase: {increase.toLocaleString('de-DE')}
-              </li>
-            ))}
-            <LineChart lineChartData={chartInputData} />
-            <BarChart barChartData={chartInputData} />
-          </InputDataList>
-          <InputDataDialog updateEnergyConsumption={updateEnergyConsumption} />
-        </MainContainer>
-      )}
+        <Route
+          path="/home/electric"
+          element={
+            <CategoryElectric
+              energyConsumptionHistory={energyConsumptionHistory}
+              dailyTotalBudget={dailyTotalBudget}
+              totalConsumption={totalConsumption}
+              setActiveChart={setActiveChart}
+              activeChart={activeChart}
+              messageText={messageText}
+              setEnergyConsumptionHistory={setEnergyConsumptionHistory}
+              updateEnergyConsumption1={updateEnergyConsumption1}
+            />
+          }
+        ></Route>
+        <Route
+          path="/home/heating"
+          element={
+            <CategoryHeating
+              energyConsumptionHistory={energyConsumptionHistory}
+              dailyTotalBudget={dailyTotalBudget}
+              totalConsumption={totalConsumption}
+              setActiveChart={setActiveChart}
+              activeChart={activeChart}
+              messageText={messageText}
+              setEnergyConsumptionHistory={setEnergyConsumptionHistory}
+              updateEnergyConsumption={updateEnergyConsumption2}
+            />
+          }
+        ></Route>
+        <Route
+          path="/home/mobility"
+          element={
+            <CategoryMobilitiy
+              energyConsumptionHistory={energyConsumptionHistory}
+              dailyTotalBudget={dailyTotalBudget}
+              totalConsumption={totalConsumption}
+              setActiveChart={setActiveChart}
+              activeChart={activeChart}
+              messageText={messageText}
+              setEnergyConsumptionHistory={setEnergyConsumptionHistory}
+              updateEnergyConsumption={updateEnergyConsumption3}
+            />
+          }
+        ></Route>
+      </Routes>
+      <Footer />
     </>
   );
 
-  function updateChart() {
-    const chartInputData = {
-      labels: energyConsumptionHistory.reverse().map(input =>
-        input.date.toLocaleDateString('en-GB', {
-          year: 'numeric',
-          month: '2-digit',
-          day: '2-digit',
-        })
-      ),
-      datasets: [
-        {
-          label: 'Consumption Data',
-          data: energyConsumptionHistory.map(input => input.increase),
-          borderColor: energyConsumptionHistory.reverse()[0].increase > dailyTotalBudget ? 'red' : 'green',
-          borderWidth: '2',
-        },
-      ],
-    };
-    return chartInputData;
-  }
-
-  function updateEnergyConsumption(inputEnergyConsumptionValue) {
-    const newInput = {
+  function updateEnergyConsumption1(inputEnergyConsumptionValue) {
```

Comment:
I recommend that you define a function in such a way that you can use it multiple times.

Reply:
Thanks for your Comment: Can you give me a hint how can I solve this in a simple way? - I do not want to lose too much time;